lib/pry/history.rb

```diff
@@ -78,8 +80,26 @@ def to_a
       @history.dup
     end
 
+    # Filter the history with the histignore options
+    # @return [Array<String>] An array containing all the lines that are not included
+    # in the histignore.
+    def filter(history)
+      history.select { |l| l unless should_ignore?(l) }
+    end
+
     private
 
+    # Check if the line match any option in the histignore [Pry.config.history.histignore]
+    # @return [Boolean] a boolean that notifies if the line was found in the histignore array.
+    def should_ignore?(line)
+      hist_ignore = Pry.config.history.histignore
+      return false if hist_ignore.nil? || hist_ignore.empty?
+
+      strings, regex = hist_ignore.partition { |w| w.is_a?(String) }
+
+      regex.any? { |r| line =~ r } || strings.include?(line)
```

Comment:
I think you can use [this trick](https://github.com/airbrake/airbrake-ruby/blob/5073d9649b49e768adfed0cf7b00f9ce955b7d6a/lib/airbrake-ruby/filters/keys_blacklist.rb#L33) to make this shorter and easier to understand.

```
regex.any? { |r| line.to_s.match(r) }
```